tuf/repository_lib.py

```diff
@@ -1531,7 +1531,8 @@ def generate_timestamp_metadata(snapshot_filename, version, expiration_date,
 
   # Retrieve the versioninfo of the Snapshot metadata file.
   snapshot_fileinfo = {}
-  length, hashes = securesystemslib.util.get_file_details(snapshot_filename)
+  length, hashes = securesystemslib.util.get_file_details(snapshot_filename,
+      tuf.settings.FILE_HASH_ALGORITHMS)
```

Comment:
I couldn't see any reason in the spec not to use the same hashing algorithms here as we do in targets?

Reply:
That's my reading as well. The more consistency in algorithm selection between roles the better, IMO.